### PR TITLE
HTML dev docs

### PR DESCRIPTION
**Pull request by coyotte508**
# What does this PR do?

This PR generates the *whole* docs inside the CI. So when the docs fail building, the CI is red instead of green.

You can see the generated html by `build_dev_documentation.yml` here: https://github.com/huggingface/doc-build-dev/tree/main/transformers/pr_15678

You can see the generated html by `build_documentation.yml` here: https://github.com/huggingface/doc-build/tree/main/transformers/doc-build-test

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Changes to make

This is a multi-repo PR, for now relying on the `kit` branch of `doc-builder`: https://github.com/huggingface/doc-builder/pull/94. When everything's ready, the `master` branch should be used instead.

Remove `@kit`:

```
pip install git+https://github.com/huggingface/doc-builder@kit -U
```

```
ref: "kit"
```

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: .github/workflows/build_dev_documentation.yml [hunk 1]
```diff
@@ -107,10 +115,24 @@ jobs:
...
         run: |
           doc-builder build transformers transformers/docs/source --build_dir doc-build-dev --notebook_dir notebooks/transformers_doc --clean --version pr_$PR_NUMBER
 
+      - name: Make HTML documentation
+        run: |
+          cd doc-builder
+          cp -R ../doc-build-dev/transformers/pr_${PR_NUMBER}/* kit/src/routes/
+          cd kit
+
+          npm install
```
Comment (Pierrci) [inline .github/workflows/build_dev_documentation.yml:128]:
```suggestion
          npm ci
```
Given how often things break w/ svelte-kit 😬

File: .github/workflows/build_documentation.yml [hunk 2]
```diff
@@ -71,13 +82,31 @@ jobs:
...
 
       - name: Make documentation
         run: |
-          doc-builder build transformers transformers/docs/source --build_dir doc-build --notebook_dir notebooks/transformers_doc --clean
+          doc-builder build transformers transformers/docs/source --build_dir doc-build-tmp --notebook_dir notebooks/transformers_doc --clean --version $branch
+          cd doc-builder
+          cp -R ../doc-build-tmp/transformers/$branch/* kit/src/routes/
+          cd kit
+
+          npm install
```
Comment (Pierrci) [inline .github/workflows/build_documentation.yml:101]:
```suggestion
          npm ci
```
(same)

**Review by sgugger — APPROVED** [review]
Very clean!

File: .github/workflows/build_dev_documentation.yml [hunk 3]
```diff
@@ -74,7 +81,7 @@ jobs:
           apt-get -y update && apt-get install -y libsndfile1-dev
 
           pip uninstall -y doc-builder
-          pip install git+https://github.com/huggingface/doc-builder -U
+          pip install git+https://github.com/huggingface/doc-builder@kit -U
```
Comment (mishig25) [inline .github/workflows/build_dev_documentation.yml:84]: note: merge after https://github.com/huggingface/doc-builder/pull/94 is merged & remove `@kit`

File: .github/workflows/build_documentation.yml [hunk 4]
```diff
@@ -46,7 +58,7 @@ jobs:
         run: |
           sudo apt-get -y update && sudo apt-get install -y libsndfile1-dev
 
-          pip install git+https://github.com/huggingface/doc-builder
+          pip install git+https://github.com/huggingface/doc-builder@kit
```
Comment (mishig25) [inline .github/workflows/build_documentation.yml:61]: note: merge after https://github.com/huggingface/doc-builder/pull/94 is merged & remove @kit

File: .github/workflows/build_dev_documentation.yml [hunk 5]
```diff
@@ -19,6 +19,7 @@ jobs:
         with:
           repository: 'huggingface/doc-builder'
           path: doc-builder
+          ref: "kit"
```
Comment (coyotte508) [inline .github/workflows/build_dev_documentation.yml:22]: Note: remove after PR https://github.com/huggingface/doc-builder/pull/94 is merged

**Pull request merged**